src/test/java/bio/terra/app/controller/SnapshotsApiControllerTest.java

```diff
@@ -0,0 +1,130 @@
+  @MockBean private SnapshotService snapshotService;
+  @MockBean private IamService iamService;
+  @MockBean private IngestRequestValidator ingestRequestValidator;
+  @MockBean private FileService fileService;
+  @MockBean private AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  @MockBean private AssetModelValidator assetModelValidator;
+
+  private static final AuthenticatedUserRequest TEST_USER =
+      AuthenticationFixtures.randomUserRequest();
+  private static final UUID SNAPSHOT_ID = UUID.randomUUID();
+  private static final String JOB_ID = "a-job-id";
+  private static final Boolean EXPORT_GCS_PATHS = false;
+  private static final Boolean VALIDATE_PRIMARY_KEY_UNIQUENESS = true;
+
+  private static final String RETRIEVE_SNAPSHOT_ENDPOINT = "/api/repository/v1/snapshots/{id}";
+  private static final String EXPORT_SNAPSHOT_ENDPOINT = RETRIEVE_SNAPSHOT_ENDPOINT + "/export";
+
+  @BeforeEach
+  void setUp() {
+    when(authenticatedUserRequestFactory.from(any())).thenReturn(TEST_USER);
+  }
+
+  @Test
+  void testExportSnapshot() throws Exception {
+    JobModel expected = new JobModel().id(JOB_ID).jobStatus(JobModel.JobStatusEnum.RUNNING);
+
+    when(snapshotService.exportSnapshot(
+            SNAPSHOT_ID, TEST_USER, EXPORT_GCS_PATHS, VALIDATE_PRIMARY_KEY_UNIQUENESS))
+        .thenReturn(JOB_ID);
+    when(jobService.retrieveJob(JOB_ID, TEST_USER)).thenReturn(expected);
+
+    String actualJson =
+        mvc.perform(
+                get(EXPORT_SNAPSHOT_ENDPOINT, SNAPSHOT_ID)
+                    .queryParam("exportGsPaths", String.valueOf(EXPORT_GCS_PATHS))
+                    .queryParam(
+                        "validatePrimaryKeyUniqueness",
+                        String.valueOf(VALIDATE_PRIMARY_KEY_UNIQUENESS)))
+            .andExpect(status().isAccepted())
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    JobModel actual = TestUtils.mapFromJson(actualJson, JobModel.class);
+    assertThat("Job model is returned", actual, equalTo(expected));
+
+    verifyAuthorizationCall(IamAction.EXPORT_SNAPSHOT);
+    verify(snapshotService)
+        .exportSnapshot(SNAPSHOT_ID, TEST_USER, EXPORT_GCS_PATHS, VALIDATE_PRIMARY_KEY_UNIQUENESS);
+  }
+
+  @Test
+  void testExportSnapshotForbidden() throws Exception {
+    IamAction iamAction = IamAction.EXPORT_SNAPSHOT;
+    mockForbidden(iamAction);
+
+    mvc.perform(
+            get(EXPORT_SNAPSHOT_ENDPOINT, SNAPSHOT_ID)
+                .queryParam("exportGsPaths", String.valueOf(EXPORT_GCS_PATHS))
+                .queryParam(
+                    "validatePrimaryKeyUniqueness",
+                    String.valueOf(VALIDATE_PRIMARY_KEY_UNIQUENESS)))
+        .andExpect(status().isForbidden());
+
+    verifyAuthorizationCall(iamAction);
+    verifyNoInteractions(snapshotService);
+  }
+
+  /** Mock so that the user does not hold `iamAction` on the dataset. */
+  private void mockForbidden(IamAction iamAction) {
```

Comment:
I ended up inlining the `mockForbidden` method as it was only called once, but I elected to pass in the `IamAction` to the other helper method since it's nice to be able to reference it in both the mock set-up and later verification.